skimage/filters/thresholding.py

```diff
@@ -335,15 +337,20 @@ def threshold_li(image):
     >>> thresh = threshold_li(image)
     >>> binary = image > thresh
     """
+    assert_nD(image, 2)
     # Requires positive image (because of log(mean))
-    offset = image.min()
-    # Can not use fixed tolerance for float image
-    imrange = image.max() - offset
-    image -= offset
+    offset = img.min()
+    if offset < 0:
```

Comment:
My feel is that this is too much trouble to avoid making a copy. I doubt this algorithm is memory-limited. 

If we do want to keep the logic, though, the condition for this should be `if offset != 0:` because _any_ nonzero offset will modify the input.